### PR TITLE
Draft: inline functions in shader code

### PR DESCRIPTION
**Pull request by Popov72**

### CODE REVIEW

File: src/Engines/Processors/shaderProcessor.ts [hunk 1]
```diff
@@ -23,7 +24,10 @@ export class ShaderProcessor {
     public static Process(sourceCode: string, options: ProcessingOptions, callback: (migratedCode: string) => void) {
         this._ProcessIncludes(sourceCode, options, (codeWithIncludes) => {
             let migratedCode = this._ProcessShaderConversion(codeWithIncludes, options);
-            callback(migratedCode);
+            let sci = new ShaderCodeInliner(migratedCode);
+            sci.debug = true;
```
Comment (deltakosh) [inline src/Engines/Processors/shaderProcessor.ts:28]: do we want to keep it in debug?

Comment (sebavan) [reply src/Engines/Processors/shaderProcessor.ts:28]: we wont even keep it in GL :-), nw I flag as resolve as this file will not be in the final PR.

File: src/Engines/Processors/shaderProcessor.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { ShaderDefineExpression } from './Expressions/shaderDefineExpression';
 import { ShaderDefineArithmeticOperator } from './Expressions/Operators/shaderDefineArithmeticOperator';
```
Comment (sebavan) [inline src/Engines/Processors/shaderProcessor.ts:9]: could you simply revert this file ??? and we ll merge it. It works beautifully well so far in WebGPU, so it should be the same in native , thanks a ton.

@syntheticmagus @CedricGuillemet @bghgary to keep on your side the integration snippet for native :-)

**Comment by Popov72** [comment]
Rollback done!

File: src/Engines/Processors/shaderProcessor.ts (hunk 2 above)
Comment (Popov72) [reply src/Engines/Processors/shaderProcessor.ts]: Done

**Pull request merged**